src/main/java/io/tiledb/TileDBCloudConnection.java

```diff
@@ -159,6 +162,20 @@ public DatabaseMetaData getMetaData() throws SQLException {
               sharedTo);
       tileDBCloudDatabaseMetadata.setArraysShared(resultShared);
     } catch (Exception e) {
+      e.printStackTrace();
+    }
+
+    // public arrays
+    if (listPublicArrays) {
+      try {
+        ArrayBrowserData resultPublic =
```

Comment:
Could we perhaps run the listings async in the backgroup to parallelize?

Reply:
yes absolutely, I will keep working on the PR

Reply:
Done